app/src/main/java/com/zeapo/pwdstore/git/BaseGitActivity.kt

```diff
@@ -90,48 +92,44 @@ abstract class BaseGitActivity : AppCompatActivity() {
         if (serverHostname.isEmpty() || !serverPort.isDigitsOnly())
             return false
 
-        val previousUrl = if (::url.isInitialized) url else ""
+        val previousUrl = url ?: ""
         val hostnamePart = serverHostname
         val pathPart = if (serverPath.startsWith('/')) serverPath else "/$serverPath"
-        url = when (protocol) {
+        val newUrl = when (protocol) {
             Protocol.Ssh -> {
                 val userPart = if (serverUser.isEmpty()) "" else "$serverUser@"
                 val portPart =
                     if (serverPort == "22" || serverPort.isEmpty()) "" else ":$serverPort"
+                if (hostnamePart.startsWith("ssh://"))
+                    hostnamePart.replace("ssh://", "")
                 // We have to specify the ssh scheme as this is the only way to pass a custom port.
-                val urlWithFreeEntryScheme = "$userPart$hostnamePart$portPart$pathPart"
-                val parsedUrl = try {
-                    URL(urlWithFreeEntryScheme)
-                } catch (_: MalformedURLException) {
-                    return false
-                }
-                if (parsedUrl.protocol == null)
-                    "ssh://$urlWithFreeEntryScheme"
-                else
-                    urlWithFreeEntryScheme
+                "ssh://$userPart$hostnamePart$portPart$pathPart"
             }
             Protocol.Https -> {
                 val portPart =
                     if (serverPort == "443" || serverPort.isEmpty()) "" else ":$serverPort"
                 val urlWithFreeEntryScheme = "$hostnamePart$portPart$pathPart"
-                val parsedUrl = try {
-                    URL(urlWithFreeEntryScheme)
-                } catch (_: MalformedURLException) {
-                    return false
-                }
-                when (parsedUrl.protocol) {
-                    null -> "https://$urlWithFreeEntryScheme"
-                    "http" -> urlWithFreeEntryScheme.replaceFirst("http:", "https:")
-                    else -> urlWithFreeEntryScheme
+                when {
+                    urlWithFreeEntryScheme.startsWith("https://") -> urlWithFreeEntryScheme
+                    urlWithFreeEntryScheme.startsWith("http://") -> urlWithFreeEntryScheme.replaceFirst("http", "https")
+                    else -> "https://$urlWithFreeEntryScheme"
                 }
             }
         }
+        try {
+            if (URI(newUrl).rawAuthority == null)
+                return false
+        } catch (ex: MalformedURLException) {
+            e(ex)
```

Comment:
Maybe add a tag here or delete it outright, I just put this on for debugging.

Reply:
Removed.